pkg/ast/common.go

```diff
@@ -37,10 +38,23 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
+type CaseConversionInfo struct {
+	dualCaseCheckEnabled   bool
+	valueIsCaseInSensitive bool
+	valueIsRegex           bool
+	IsString               bool
+	originalColValue       interface{}
+}
+
+func (cci *CaseConversionInfo) ShouldSearchWithOriginalCase() bool {
+	return cci.IsString && cci.dualCaseCheckEnabled && cci.valueIsCaseInSensitive && !cci.valueIsRegex
+}
```

Comment:
okay, so may be this name is confusing I guess, but I named the original string send by the user as `originalColValue` and that is why I said `ShouldSearchWithOriginalCase`. But may be you can suggest me a better name here or may be I should write some comment above this function. 

So basically what this function is mainly saying is should I create / assign / maintain the `originalColValue` across the `AstNode`.  

And as for your question why it returns `true` only if `cci.valueIsCaseInSensitive == true` because, at the parser level in `spl.peg`, I am directly the assigning the search value to be lower case string (by default) or the original query string (if CASE) and then I am also assigning the `originalColValue` to the string given by the user. See this part of the code: https://github.com/siglens/siglens/pull/1497/files#diff-b61669c2b759d47b39447cac5ea6830f00f70a4193f64479ae5a4d094206a367R3899-R3913

And since the `colValue` variable will already have the string that which we need to do the search or comparison. And when search is case-sensitive, the `colValue` will already have the original string sent by the user and so there is no need to search with `originalColValue` (Also both `colValue` and `originalColValue` will be same). 

But when the search is case-insensitive, then the `colValue` will have lower case string and the `originalColValue` will have the original string given in the query. And so only when `cci.valueIsCaseInSensitive == true`, we also create the search with `originalColValue`.

Reply:
This function: `ShouldSearchWithOriginalCase` will be renamed to `ShouldAlsoSearchWithOriginalCase`